### ISSUE TITLE
Changing the order of files in a project needs a restart to be recognized in the (Pandoc) export

### ISSUE THREAD

**Issue by Mela** [Zettlr/Zettlr#4912]
### Description

I'm using numbers in the YAML header title to put Zettlr documents in a project into the order I need and I want to see in the output. 
This works mostly quite well for me. 

Except if I reorder my files or if I add a new file, the order is not recognized by the export. New documents are always last and older documents stay on their prior spot. 

I need to quit Zettlr and reopen, to have the new order recognized. 

![Bildschirmfoto 2024-01-22 um 19 49 42](https://github.com/Zettlr/Zettlr/assets/361364/330500d4-bf58-4583-ae72-5a340fb9f603)

![Bildschirmfoto 2024-01-22 um 19 51 32](https://github.com/Zettlr/Zettlr/assets/361364/d4bdfc5c-d26f-47e2-bee2-2f45b5efab56)

### Reproducing

Make a directory into a project. Add some documents.

Add YAML headers to the documents. 

Enter document titles like: 

100 — Blabla
200 — Blabla
300 — Blabla
400 — Blabla
500 — Blabla
600 — Blabla
700 — Blabla
800 — Blabla
900 — Blabla

Add some text to the documents.

Export to PDF

Now change the title from 900 — Blabla to 150 — Blabla.  

Export to PDF 

900 — Blabla will still be the last document in the export. Only after quitting Zettlr and restarting it, it will be at second place. 

### Zettlr Version

Stable (most recent version)

### Specify version

3.0.3

### Installation Method

From the Website or GitHub

### Your Platform

- [ ] Windows
- [X] macOS
- [ ] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Sonoma 14.2.1 (23C71)

**Comment by lcsolano** [comment]
I confirm this in 3.0.2. Linux .deb.

I just got used to restart as a workaround.

**Comment by nathanlesage** [comment]
Okay, a restart shouldn't be necessary, I'll investigate, as I already have a suspicion where the issue may be!

**Comment by nathanlesage** [comment]
Superseded by #4966
